### PR TITLE
Bug 1206695 - Reuse crypto keys, r=ferjm

### PR DESCRIPTION
**Pull request by michielbdejong**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=3c70105d162a1baf47761b868bb7bf2aa3369032

File: apps/sync/test/unit/sync-engine/kinto-mock.js [hunk 1]
```diff
@@ -8,6 +8,7 @@
 
 var Kinto = (function() {
   var mockProblem =  null;
+  var listData = {};
```
Comment (michielbdejong) [inline apps/sync/test/unit/sync-engine/kinto-mock.js]: I was previously keeping this data inside the mocked Collection object, but now keeping it globally here, which is a slightly more realistic simulation of how Kinto.js persists data in IndexedDB.

File: apps/sync/test/unit/sync-engine/kinto-mock.js [hunk 2]
```diff
@@ -213,8 +230,12 @@ empting to fetch resource.`));
     };
   };
 
+  Kinto.syncCount = {};
```
Comment (michielbdejong) [inline apps/sync/test/unit/sync-engine/kinto-mock.js]: This is a static property, so counts add up globally - doesn't matter how often you instantiate `new Kinto()` or how often you call `myKintoInstance.collection()`.

File: apps/sync/js/sync-engine/syncengine.js [hunk 3]
```diff
@@ -305,15 +312,13 @@ ting to fetch resource.`) {
       }).then(() => {
         return this._syncCollection('meta');
       }).then(() => {
-        return this._getItem('meta', 'global');
+        return this._getItem('meta', 'global', false);
```
Comment (ferjm) [inline apps/sync/js/sync-engine/syncengine.js]: `return this._getItem('meta', 'global', false /* syncIfNeeded */);` or simply `return this._getItem('meta', 'global');`

File: apps/sync/js/sync-engine/syncengine.js [hunk 4]
```diff
@@ -305,15 +312,13 @@ ting to fetch resource.`) {
...
       }).then(metaGlobal => {
         if (!this._storageVersionOK(metaGlobal)) {
           return Promise.reject(new SyncEngine.UnrecoverableError(`Incompatible\
  storage version or storage version not recognized.`));
         }
-        return this._syncCollection('crypto');
-      }).then(() => {
-        return this._getItem('crypto', 'keys');
+        return this._getItem('crypto', 'keys', true);
```
Comment (ferjm) [inline apps/sync/js/sync-engine/syncengine.js]: `return this._getItem('crypto', 'keys', true /* syncIfNeeded */);`

File: apps/sync/test/unit/sync-engine/syncengine_test.js [hunk 5]
```diff
@@ -9,6 +9,7 @@
   FxSyncWebCrypto,
   Kinto,
   requireApp,
+  setup,
```
Comment (ferjm) [inline apps/sync/test/unit/sync-engine/syncengine_test.js]: I don't know why we need to declare mocha related globals here. This is not happening with other apps. Could you file a bug to investigate it, please? Thanks!

Comment (michielbdejong) [inline apps/sync/test/unit/sync-engine/syncengine_test.js]: Opened https://bugzilla.mozilla.org/show_bug.cgi?id=1213882

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=d79bc5d705b0a8fbda55243e4c1efe1e5780e491

**Pull request merged**